### ISSUE TITLE
Reports: Filtering for Instructor gives no results (mbox|account-Agent)

### ISSUE THREAD

**Issue by psacher** [LearningLocker/learninglocker#655]
Hi, 

*one little thing first:
I have LL v1.3.9 installed and I can't update to the newest version at the moment because I don't have access to the servers directly. So if this bug is fixed already, sorry for reporting it. I haven't read anything about fixing this in the Logs, so maybe it's still present in the newer versions.*

I stumbled across a problem, when Filtering for Instructor in the Reporting-Tool.

It doesn't matter, how I define the Agent-Object that represents the Instructor, I don't get any results.
:(

I tried defining the Instructor as Agent with mbox, or Agent with account. None of it worked.
mbox looks better, because its not "undefined", but the Report Results always stay empty and no matching Statements are found.

Below is the instructor/context part of my statements and the Result, what LearningLocker-Reporter does with it in the Reporting tool.

**Instructor declared *mbox* in Statement:**
```javascript
"context": {
        "instructor": {
            "objectType": "Agent",
            "name": "Patrick",
            "mbox": "mailto:patrick@example.com"
        },
}
```
![instructor-mbox](https://cloud.githubusercontent.com/assets/10605556/7942987/0657cf8a-0962-11e5-87ea-2328b43b07ee.jpg)

**Instructor declared as *account* in Statement:**
```javascript
"context": {
        "instructor": {
            "objectType": "Agent",
            "name": "Patrick",
            "account": {
                "name": "patrick",
                "homePage": "http://example.com"
            }
        },
}
```
![instructor-account](https://cloud.githubusercontent.com/assets/10605556/7942725/683b7cda-0960-11e5-9193-08fc47c65c1f.jpg)

I would love to declare the Instructor with account and I can't find anything in the xAPI-Specs, that forbids this.
So is there a bug or are my Statements wrong?
Thx for your time.

**Comment by ryansmith94** [comment]
Hi thanks for reporting the issue. In future please use our [templates](https://github.com/LearningLocker/learninglocker/blob/develop/contributing.md) when reporting an issue.

This indeed a bug in the current version too. I'll try to look into it when I get an opportunity.

**Comment by ryansmith94** [comment]
In the meantime, you can use our [aggregation API](http://docs.learninglocker.net/statements_api/#aggregate).

**Comment by psacher** [comment]
Thanks for your help and answer.
Good luck fixing it ^^

**Comment by ryansmith94** [comment]
You're welcome. Thanks :+1:

**Comment by ryansmith94** [comment]
Currently working on this by the way. It will take some time though.

**Comment by ryansmith94** [comment]
Hi @psacher, this issue has now been fixed in #667. This should be released with version 1.5.2. Sorry it took so long, it's a difficult issue because the filtering for agents is a little quirky due to the different identifiers that can be used.
